services/asset_bundle/flx_asset_bundle.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "base/task_runner.h"
+#include "base/files/file_path.h"
+#include "base/memory/weak_ptr.h"
+#include "mojo/public/cpp/bindings/interface_request.h"
+#include "mojo/public/cpp/bindings/strong_binding.h"
+#include "mojo/public/cpp/environment/async_waiter.h"
+#include "mojo/services/asset_bundle/interfaces/asset_bundle.mojom.h"
+#include "third_party/zlib/contrib/minizip/unzip.h"
+
+namespace mojo {
+namespace asset_bundle {
+
+// An implementation of AssetBundle that serves assets directly out of an
+// FLX (ZIP) file.
+class FlxAssetBundle : public AssetBundle {
```

Comment:
Is there anything specific to FLX files in this code?  I might just call this a FileAssetBundle.  Conceptually the asset_bundle service shouldn't know anything about Flutter (or FLX files for that matter).